### DISCUSSION TITLE
Does runtime destroy agent instance?

### DISCUSSION THREAD

**Discussion by vincent-pli** [microsoft/autogen#4207] [Q&A] [answered]
I took a glance at the code of `singlethreadagentruntime`, I found it will create agent instance if it not existed, but when the instance been destoried?

**Comment by ekzhu** [comment, marked as answer]
Good observation! It currently doesn't. But it need to be designed together with other stateful behaviors like runtime serialization. We are aiming to get to this after the initial 0.4 release.
